R/finalize_plot.R

```diff
@@ -315,29 +338,113 @@ finalize_plot <- function(input_plot = NULL,
                     col = cmapplot_globals$colors$blackish)
   )
 
+
+  # Function to create plot object with left aligned legend on top
+  buildChart <- function(p,l_margins,p_margins,l_indent,l_height,p_height) {
+
+    # Add left indent to plot margins
+    updated_l_margins <- c(l_margins[1:3],l_margins[4] + l_indent)
+
+    # Create a ggplot table of the updated plot element, including a left-aligned legend
+    gt <- ggplot2::ggplot_gtable(
+      ggplot2::ggplot_build(p + theme(legend.position = "left",
+                                      legend.justification = "left",
+                                      legend.direction = "horizontal",
+                                      plot.title = element_blank(),
+                                      plot.margin = grid::unit(p_margins,"bigpts"),
+                                      legend.margin = margin(t = updated_l_margins[1],
+                                                             r = updated_l_margins[2],
+                                                             b = updated_l_margins[3],
+                                                             l = updated_l_margins[4],
+                                                             "bigpts")
+                                      )
+                            )
+      )
+
+    # Extract the legend
+    l <- gtable::gtable_filter(gt, "guide-box")[[1]]
```

Comment:
I'm a little bit confused here. if ggpubr::get_legend can extract a legend from a ggplot, why the reliance here on ggplot_build, gtable_filter, etc? I haven't touched the code at all, but it seems like get_legend could [do the extraction](https://rpkgs.datanovia.com/ggpubr/reference/get_legend.html) with more simply?

Reply:
I had used that method because it was what that one you sent had, and it worked (!), but yes, it turns out that you can eliminate a couple calls and make it a tad simpler (although still complicated) with the get_legend function. I've adjusted.

Reply:
And, doing it the new way should improve some debugging functions, which is a nice side benefit.